internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -129,13 +129,21 @@ func (connector *SingleUseConnector) Connect(
 	clientBuffer := mssql.NewTdsBuffer(bufferSize, connector.clientConn)
 	_, err := mssql.ReadPreloginWithPacketType(clientBuffer, mssql.PackPrelogin)
 	if err != nil {
-		return connector.sendError("failed to read prelogin request: %s", err)
+		newError := fmt.Errorf(
```

Comment:
The errors.Wrap() function might be handy here:
```
import "github.com/pkg/errors"
```
and:
```
    newError := errors.Wrap(err, "failed to read prelogin request")
```
and similarly for other error wrappings below.

Reply:
Or could you redefine connector.sendError() to do the wrapping?:
```
func (connector *SingleUseConnector) sendError(err error, wrapStr string) {
   . . .
}
```

Reply:
wrappers the errors themselves so that the wrapped error could be returned